drivers/driver.go

```diff
@@ -45,8 +43,19 @@ type Driver interface {
 	//
 	// Run() MUST monitor the context. task cancellation is indicated by
 	// cancelling the context.
-	// In addition, Run() should respect the task's timeout.
-	Run(ctx context.Context, task ContainerTask) (RunResult, error)
+	Run(ctx context.Context) (RunResult, error)
+}
+
+type Driver interface {
+	// Prepare can be used in order to do any preparation that a specific driver
+	// may need to do before running the task, and can be useful to put
+	// preparation that the task can recover from into (i.e. if pulling an image
+	// fails because a registry is down, the task doesn't need to be failed).  It
+	// returns a cookie that can be used to execute the task.
+	// Callers should Close the cookie regardless of whether they run it.
+	//
+	// The returned cookie should respect the task's timeout when it is run.
```

Comment:
the timeout is done in the cookie? i'm confused what this means for a caller, do you mean s/should/will/ ?